### PR TITLE
Comprehension internal tools/ RegexRules page

### PR DESCRIPTION
**Pull request by eadams17**
## WHAT
add a view for RegexRules in the internal tools. This also adds a `RuleViewForm` component that will eventually be used for all of the different rule types CRUD actions and view.

## WHY
so Curriculum can more easily navigate regex rules

## HOW
added a `RegexRulesRouter` to handle the new index and CRUD views

### Screenshots
(If applicable. Also, please censor any sensitive data)
Screen Shot 2021-04-30 at 4 34 24 PM

### Notion Card Links
(Please provide links to any relevant Notion card(s) relevant to this PR.)
https://www.notion.so/quill/New-Comprehension-Pages-Regex-Rules-page-and-Plagiarism-Rules-Page-d66141aa5b19496aa8adec012c3a9c1c

PR Checklist | Your Answer
------------ | -------------
Have you added and/or updated tests? |  Yes
Have you deployed to Staging? | Not yet - deploying on Monday
Self-Review: Have you done an initial self-review of the code below on Github? | Yes
Design Review: If applicable, have you compared the coded design to the mockups? | Yes

### CODE REVIEW

**Review by cissyxyu — APPROVED** [review]

File: services/QuillLMS/client/app/bundles/Staff/components/comprehension/configureRules/ruleViewForm.tsx [hunk 1]
```diff
@@ -0,0 +1,261 @@
...
+      </section>
+      <form className="semantic-rule-form">
+        <RuleGenericAttributes
+          autoMLParams={autoMLParams}
+          concepts={conceptsData ? conceptsData.concepts : []}
+          errors={errors}
+          isAutoML={true}
+          isUniversal={isUniversal}
+          ruleConceptUID={ruleConceptUID}
+          ruleDescription={ruleDescription}
```
Comment (emilia-friedberg) [inline services/QuillLMS/client/app/bundles/Staff/components/comprehension/configureRules/ruleViewForm.tsx]: here too

File: services/QuillLMS/client/app/bundles/Staff/components/comprehension/configureRules/ruleViewForm.tsx [hunk 2]
```diff
@@ -0,0 +1,261 @@
...
+          setRuleType={setRuleType}
+        />
+        {isSemantic && <RuleSemanticAttributes
+          errors={errors}
+          ruleLabelName={ruleLabelName}
+          ruleLabelNameDisabled={!!(label && label.name)}
+          ruleLabelStatus={ruleLabelStatus}
+          setRuleLabelName={setRuleLabelName}
+        />}
+        {ruleType && ruleType.value === 'plagiarism' && <RulePlagiarismAttributes
```
Comment (emilia-friedberg) [inline services/QuillLMS/client/app/bundles/Staff/components/comprehension/configureRules/ruleViewForm.tsx]: we should probably have and be using constants for things like "plagiarism"

File: services/QuillLMS/client/app/bundles/Staff/components/comprehension/regexRules/regexRulesIndex.tsx [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+  const addRulesBased3Link = <Link to={{ pathname: `/activities/${activityId}/regex-rules/new`, state: { ruleType: 'rules-based-3' }}}>Add Typo Regex Rule</Link>;
+  const rulesBased1Rows = getFormattedRows(rulesBased1Data);
+  const rulesBased2Rows = getFormattedRows(rulesBased2Data);
+  const rulesBased3Rows = getFormattedRows(rulesBased3Data);
+
+  return(
+    <div className="rules-container">
+      <section className="rules-based-section">
+        <button className="quill-button fun primary contained add-rule-button" type="submit">{addRulesBased1Link}</button>
+        {renderTable(rulesBased1Rows, 'Sentence Struture')}
```
Comment (emilia-friedberg) [inline services/QuillLMS/client/app/bundles/Staff/components/comprehension/regexRules/regexRulesIndex.tsx]: I think this should be "Sentence Structure"

File: services/QuillLMS/client/app/bundles/Staff/components/comprehension/regexRules/regexRulesIndex.tsx [hunk 4]
```diff
@@ -0,0 +1,156 @@
...
+
+    // get cached activity data to pass to rule
+    const { data: activityData } = useQuery({
+      queryKey: [`activity-${activityId}`, activityId],
+      queryFn: fetchActivity
+    });
+
+  const { data: rulesBased1Data } = useQuery({
+    // cache rules data for updates
+    queryKey: [`rules-${activityId}`, activityId, null, 'rules-based-1'],
```
Comment (emilia-friedberg) [inline services/QuillLMS/client/app/bundles/Staff/components/comprehension/regexRules/regexRulesIndex.tsx]: Isn't this always going to be the same data as the other two? Also, I think we should change `rulesBased1` etc to be `sentenceStructureRules` or whatever - it's hard to tell what's going on when they're just numbers.

**Review by emilia-friedberg — APPROVED** [review]
LGTM, just some tweaks requested. Annoyingly, I think there are a couple of places you'll have to update `description` to `note` when this rename PR goes out this afternoon - I'm sorry about that.

File: services/QuillLMS/client/app/bundles/Staff/components/comprehension/configureRules/ruleViewForm.tsx (hunk 2 above)
Comment (eadams17) [reply services/QuillLMS/client/app/bundles/Staff/components/comprehension/configureRules/ruleViewForm.tsx]: good call, I'll add it as a const

File: services/QuillLMS/client/app/bundles/Staff/components/comprehension/regexRules/regexRulesIndex.tsx (hunk 4 above)
Comment (eadams17) [reply services/QuillLMS/client/app/bundles/Staff/components/comprehension/regexRules/regexRulesIndex.tsx]: So this is what the actual rule type is in the DB, this only fetches Sentence Structure regex rules-- the url endpoint ends up looking like this: `http://quill.org/api/v1/comprehension/rules?rule_type=rules-based-1`

and this is what it looks like in the `RulesController`:
```
    def index
      @rules = Comprehension::Rule
      @rules = @rules.joins(:prompts_rules).where(comprehension_prompts_rules: {prompt_id: params[:prompt_id]}) if params[:prompt_id]
      @rules = @rules.where(rule_type: params[:rule_type]) if params[:rule_type]

      render json: @rules.all
    end

**Pull request merged**